### PR TITLE
fix(iroh-net): enforce storing a single derp region per peer

### PR DESCRIPTION
**Pull request by divagant-martian**
## Description

The Endpoint contained `SendAddr` instead of `ip`, `port` pairs. This is problematic for a couple of reasons:
- it makes it harder to reason about direct addresses and relay addresses
- makes it harder to properly update the derp-related information
- in principle, we shouldn't be storing multiple derp regions per peer

So this PR enforces storing a single derp region per peer. Some relevant note with these changes:

## Notes & open questions

- this proves we never ping a peer via their relay address. I left some TODOs that I intend to address in a later PR when it's clear how we should be doing this
- I also noticed updating a peer's region is different between "updating from node" to receiving a ping. Maybe this is the right thing to do but it's also peculiar. If a peer pinged us via relay we update their derp region, if we add a ticket or receive info from a gossip peer exchange we only update the region if we don't have any. To me this makes sense, but again, it draws my attention. I left a comment about it in the code
Basically TLDR, next step is adding ping via relays with some appropriate tests. This should at a minimum solve the second part of #1576 

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

**Comment by dignifiedquire** [comment]
this looks pretty good, what is missing for this to be ready?

**Comment by divagant-martian** [comment]
@dignifiedquire was built on top of #1606 so was waiting for that. Otherwise I think it's ready. I'll update the description with the relevant info

**Review by dignifiedquire — APPROVED** [review]
